File: zenhacks/strangegrid.c
// Matt Piazza
// 27 FEB 2015
// solution to: https://www.hackerrank.com/contests/zenhacks/challenges/strange-grid

#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

unsigned long getvoffset(unsigned long row){
   row = row - 1;
   return (row - (row % 2)) * 5 + (row % 2);
   // case: r = 2 -> 1 - 1 * 5 + 1 = 1 offset on row 2
   // or r = 3 -> 2 - 0 * 5 + 0 = 10 offset on row 3
} 

int main() {
   // odd rows have even nums
   // offsets by 10 every two rows
   int c;
   unsigned long r, num = 0, voffset = 0;
   scanf("%ld %d\n", &r, &c);
   voffset = getvoffset(r);
   num = voffset + (c - 1) * 2;
   printf("%ld\n", num);
   return 0;
}

File: math/strangegrid.go
// solution to https://www.hackerrank.com/challenges/strange-grid

package main

import "fmt"

func main() {
	var col, row int64
	fmt.Scanf("%d %d", &row, &col)
	fmt.Println(compute(row, col))
	//Enter your code here. Read input from STDIN. Print output to STDOUT
}

func compute(row, col int64) int64 {
	base := int64(((row - 1) / 2) * 10)
	var offset int64
	if row%2 == 0 {
		// even rows have odd numbers
		offset = col*2 - 1

	} else {
		// odd rows have even numbers
		offset = col*2 - 2 // minus 2 since column numbers are base 1 :(
	}
	return base + offset
}
